File: FinalProject/Models/Keep.cs
namespace FinalProject.Models
{
    public class Keep
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string img { get; set; }
        public int views { get; set; }
        public int kept { get; set; }
        public string CreatorId { get; set; }
        public Account Creator { get; set; }

    }

    public class KeepsInVault : Keep
    {
        public int VaultId { get; set; }
        public int VaultKeepId { get; set; }
    }
}

File: FinalProject.client/src/models/Keep.js
import { Account } from "./Account.js"

export class Keep {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.creatorId = data.creatorId
        this.creator = new Account(data.creator)
    }
}

export class KeepInVault extends Keep {
    constructor(data) {
        super(data)
        this.vaultKeepId = data.vaultKeepId
    }
}
